src/wolfsentry_util.c

```diff
@@ -812,8 +812,10 @@ WOLFSENTRY_API wolfsentry_errcode_t wolfsentry_init_thread_context(struct wolfse
          * the held lock can be safely locked recursively from within the
          * interrupt context.
          */
-        if (thread_context->id == (wolfsentry_thread_id_t)((uintptr_t)WOLFSENTRY_THREAD_NO_ID - 0xffffff))
-            WOLFSENTRY_ATOMIC_INCREMENT(fallback_thread_id_counter, 0xffffff);
+        if (thread_context->id == (wolfsentry_thread_id_t)((uintptr_t)WOLFSENTRY_THREAD_NO_ID - 0xffffff)) {
+            (void)WOLFSENTRY_ATOMIC_INCREMENT(fallback_thread_id_counter, 0xffffff);
+        }
+        (void)fallback_thread_id_counter;
```

Comment:
this `(void)` can't be needed -- `WOLFSENTRY_ATOMIC_INCREMENT()` falls through to a regular C increment on singlethreaded builds, so there won't ever be an unused-variable warning.